src/Avalonia.Controls/Utils/BorderRenderHelper.cs

```diff
@@ -85,19 +110,22 @@ public void Render(DrawingContext context, Size size, Thickness borders, CornerR
             }
             else
             {
-                var borderThickness = borders.Top;
-                var top = borderThickness * 0.5;
-
+                var borderThickness = _borderThickness.Top;
                 IPen pen = null;
 
                 if (borderThickness > 0)
                 {
                     pen = new Pen(borderBrush, borderThickness);
                 }
 
-                var rect = new Rect(top, top, size.Width - borderThickness, size.Height - borderThickness);
+                var rrect = new RoundedRect(new Rect(_size), _cornerRadius.TopLeft, _cornerRadius.TopRight,
```

Comment:
You need to calculate the rdii in case the thickness is not uniform. This only works for uniform thickness.